utcp.h

```diff
@@ -43,14 +43,18 @@ struct utcp_connection;
 #define UTCP_SHUT_WR 1
 #define UTCP_SHUT_RDWR 2
 
+// for the send callback
+#define UTCP_ERROR -1
+#define UTCP_WOULDBLOCK -2
+
 typedef bool (*utcp_pre_accept_t)(struct utcp *utcp, uint16_t port);
 typedef void (*utcp_accept_t)(struct utcp_connection *utcp_connection, uint16_t port);
 
 typedef ssize_t (*utcp_send_t)(struct utcp *utcp, const void *data, size_t len);
-typedef ssize_t (*utcp_recv_t)(struct utcp_connection *connection, const void *data, size_t len);
+typedef void (*utcp_recv_t)(struct utcp_connection *connection, const void *data, size_t len);
 
 typedef void (*utcp_ack_t)(struct utcp_connection *connection, size_t len);
-typedef void (*utcp_poll_t)(struct utcp_connection *connection, size_t len);
+typedef int (*utcp_poll_t)(struct utcp_connection *connection, size_t len);
```

Comment:
@a-krebs @gsliepen agreed but here I'd like to have feedback first if it's ok to change all the function's return types to return the error code and not just true or false
ofc the error code could also be retrieved by errno after the calls, however then we'd have to assure that on each possible error it is set so it doesn't mess with previously set error from earlier calls

I pretty much prefer to leave errno calls for system call error handling and immediately retrieve it where it occured and then somehow pass it back